### ISSUE TITLE
"Assembly has reference to non-existent assembly 'UnityEngine.TestRunner'" compiler error

### ISSUE THREAD

**Issue by rfadeev** [Unity-Technologies/BuildReportInspector#8]
Commit id: 84cbe0e6
Open `BuildReportInspector/TestProject` with Unity and observe two compilation errors in console:

```
Assembly has reference to non-existent assembly 'UnityEngine.TestRunner' (Packages/com.unity.build-report-inspector/Tests/Editor/Mobile/Unity.BuildReportInspector.Editor.Tests.Mobile.asmdef)
Assembly has reference to non-existent assembly 'UnityEngine.TestRunner' (Packages/com.unity.build-report-inspector/Tests/Editor/Unity.BuildReportInspector.Editor.Tests.asmdef)
```

Does not reproduce on master branch commit 3039f046 adding package to empty Unity project.

Reproduced with Unity 2018.4.6f1.
Not reproduced with Unity 2019.3.0f6.

**Comment by Wilfrid-Unity** [comment]
Thanks @rfadeev for reporting this.

@iRunn3r This sounds related to branch [mobile](https://github.com/Unity-Technologies/BuildReportInspector/commits/mobile) only. Do you see how to avoid that?
By the way, branch [mobile](https://github.com/Unity-Technologies/BuildReportInspector/commits/mobile) is still under development, right? Or would you say it is quite stable already?

**Comment by rfadeev** [comment]
@Wilfrid-Unity No problem! Since pull request #7 with `mobile` branch is targeting master, it's a potential problem. If it's a known problem, then this issue is false alarm and I'm fine to close it.

This can be fixed by having "Test Assemblies" toggle on in inspector for those problematic asmdef files. Then asmdef json would have:
* `"references"` array without`"UnityEngine.TestRunner"` and `"UnityEditor.TestRunner"` elements
* `"optionalUnityReferences"` array with single `"TestAssemblies"` element
* `"precompiledReferences"` array without `"nunit.framework.dll"` element

**Comment by vaidas-unity** [comment]
Thank you for the fix suggestion, @rfadeev, I have applied it at [c210103](https://github.com/Unity-Technologies/BuildReportInspector/commit/c210103687071533d008b069ab1f27de66e8cbf4).
